core/src/main/scala/quasar/destination/avalanche/AvalancheDestination.scala

```diff
@@ -0,0 +1,274 @@
+    localDateTimeFormat = DateTimeFormatter.ofPattern("uuuu-MM-dd HH:mm:ss.SSS"),
+    localDateFormat = DateTimeFormatter.ofPattern("uuuu-MM-dd"),
+    localTimeFormat = DateTimeFormatter.ofPattern("HH:mm:ss.SSS"))
+
+  def destinationType: DestinationType =
+    AvalancheDestinationModule.destinationType
+
+  def sinks: NonEmptyList[ResultSink[F]] = NonEmptyList(csvSink)
+
+  private val csvSink: ResultSink[F] = ResultSink.csv[F](IngresRenderConfig) {
+    case (path, columns, bytes) =>
+      for {
+        tableName <- Stream.eval(ensureValidTableName(path))
+
+        cols <- ApplicativeError[Stream[F, ?], Throwable].fromEither(
+          ensureValidColumns(columns).leftMap(new RuntimeException(_)))
+
+        freshName <- Stream.eval(upload(bytes))
+
+        _ <- Stream.eval(copy(tableName, freshName, cols)).onFinalize(deleteBlob(freshName))
+
+      } yield ()
+  }
+
+  private def deleteBlob(freshName: FileName): F[Unit] =
+    for {
+      _ <- refreshToken
+
+      containerURL <- refContainerURL.get
+
+      deleteService = AzureDeleteService.mk[F](containerURL.value)
+
+      _ <- deleteService(BlobPath(List(PathElem(freshName.value))))
+    } yield ()
+
+  private def copy(tableName: TableName, freshName: FileName, cols: NonEmptyList[Fragment])
+      : F[Unit] = {
+
+    val dropQuery = dropTableQuery(tableName).update
+
+    val createQuery = createTableQuery(tableName, cols).update
+
+    val loadQuery = copyQuery(tableName, freshName).update
+
+    for {
+      _ <- debug(s"Table creation query:\n${createQuery.sql}")
+
+      _ <- debugRedacted(s"Load query:\n${loadQuery.sql}")
+
+      _ <- debug(s"Drop table query:\n${dropQuery.sql}")
+
+      load = Sync[ConnectionIO].bracket(createStatement)(st =>
+        for {
+          _ <- Sync[ConnectionIO].delay(st.execute(loadQuery.sql))
+          count <- Sync[ConnectionIO].delay(st.getUpdateCount)
+        } yield count)(st => Sync[ConnectionIO].delay(st.close))
+
+      count <- (dropQuery.run *> createQuery.run *> load).transact(xa)
+
+      _ <- debug(s"Finished load")
+
+      _ <- debug(s"Load result count: $count")
+
+      _ <- debug(s"Finished table copy")
+
+    } yield ()
+  }
+
+  private def upload(bytes: Stream[F, Byte]): F[FileName] =
+    for {
+      freshNameSuffix <- Sync[F].delay(UUID.randomUUID().toString)
+      freshName = s"reform-$freshNameSuffix"
+
+      _ <- refreshToken
+
+      containerURL <- refContainerURL.get
+
+      put = AzurePutService.mk[F](containerURL.value)
+
+      _ <- debug(s"Starting upload of $freshName")
+
+      _ <- put((BlobPath(List(PathElem(freshName))), bytes))
+
+      _ <- debug(s"Finished upload of $freshName")
+
+    } yield FileName(freshName)
+
+  // Ingres accepts double quotes as part of identifiers, but they must
+  // be repeated twice This implies identifiers with an even number of
+  // double quotes are valid and those with an odd number of double
+  // quotes are invalid.
```

Comment:
Hmm, this is not entirely true: the double quotes must also be consecutive. Eg `"a "string" with quotes"` would pass the test but is not valid and would allow for sql injection. 

I would approach this the other way around: if `a "string" with quotes` is specified as table name by the user, then we should generate sql like `SELECT from "a ""string"" with quotes"`, i.e. we generate a delimited identifier with escaped double quotes.

BTW Single quotes are valid characters for delimited identifiers so if the table name is specified as `I'm using a single quote` then we can safely generate that as `"I'm using a single quote"`

Reply:
Yes. This is the problem I need to solve today. This is not done yet